api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -617,112 +597,58 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
   }
 
   @Override
-  public ResponseEntity<FileDetail> renameNotebook(String workspace, String workspaceName,
-      NotebookRename rename) {
-    String newName = rename.getNewName();
-    if (!newName.matches("^.+\\.ipynb")) {
-      newName = newName + ".ipynb";
-    }
-    FileDetail fileDetail = notebookCloneOperation(workspace, workspaceName, rename.getName(),
-        newName);
-    notebookDeleteOperation(workspace, workspaceName, rename.getName());
-    return ResponseEntity.ok(fileDetail);
-  }
-
-  @Override
-  public ResponseEntity<FileDetail> cloneNotebook(String workspace, String workspaceName,
-      String notebookName) {
-    String newName = notebookName.replaceAll("\\.ipynb", " ") + "Clone.ipynb";
-    FileDetail fileDetail = notebookCloneOperation(workspace, workspaceName, notebookName, newName);
-    return ResponseEntity.ok(fileDetail);
+  public ResponseEntity<List<FileDetail>> getNoteBookList(String workspaceNamespace,
+      String workspaceId) {
+    List<FileDetail> fileList = notebooksService.getNotebooks(workspaceNamespace, workspaceId);
+    return ResponseEntity.ok(fileList);
   }
 
   @Override
   public ResponseEntity<EmptyResponse> deleteNotebook(String workspace, String workspaceName,
       String notebookName) {
-    notebookDeleteOperation(workspace, workspaceName, notebookName);
+    notebooksService.deleteNotebook(workspace, workspaceName, notebookName);
     return ResponseEntity.ok(new EmptyResponse());
   }
 
-  private FileDetail notebookCloneOperation(String workspace, String workspaceName,
-      String notebookName, String newName) {
-    NotebookOpSetup opDto = new NotebookOpSetup(workspace, workspaceName, notebookName, newName);
-    FileDetail fileDetail = new FileDetail();
-    cloudStorageService.copyBlob(opDto.blobId, opDto.newBlobId);
-    fileDetail.setName(newName);
-    fileDetail.setPath(opDto.fullPath);
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    fileDetail.setLastModifiedTime(now.getTime());
-    userRecentResourceService
-        .updateNotebookEntry(opDto.workspaceId, opDto.userId, opDto.fullPath, now);
-    return fileDetail;
-  }
+  @Override
+  public ResponseEntity<FileDetail> copyNotebook(String fromWorkspaceNamespace, String fromWorkspaceId,
```

Comment:
Shouldn't this be called `duplicateNotebook`? I don't think this should be a different method from clone. If the plan is to remove the other method in a follow-up, that's fine (please add a TODO if so).

Reply:
I can sync up with product on this but in my mind, Copy means an entity can be copied into some other container while Clone/Duplicate means creating a new version within the same container.

Reply:
I think this is entirely an engineering call, product will probably not be looking at our REST APIs. I think that functionally, duplicating within a workspace is a proper subset of duplicating across workspaces, with no other semantic distinction. Product can call these buttons two different things if they want, we don't need our API to mirror that.

Reply:
True. Are you saying we should just remove `cloneNotebook` and use `copyNotebook` in its place?

Reply:
Yep - could also be in a follow-up PR if you prefer.

Reply:
I'll do it in a follow up. It's going to be hard to pick out the changes in this PR since its already so big. Also there's a good chunk of changes that need to be made across the stack.

https://precisionmedicineinitiative.atlassian.net/browse/RW-2551